quantities/quantities_body.hpp

```diff
@@ -9,11 +9,34 @@
 #include <string>
 
 #include "base/macros.hpp"
+#include "base/not_constructible.hpp"
 
 namespace principia {
 namespace quantities {
 namespace internal_quantities {
 
+using base::not_constructible;
+
+// A helper class to serialize dimension exponents.  Any change here is
+// potentially save-incompatible.
+class ExponentSerializer : not_constructible {
+ public:
+  // Returns true if the exponent is in the range that we can serialize.
+  static constexpr bool IsSerializable(std::int64_t exponent);
+
+  // Returns the serialized representation of the exponent.  |position| is the
+  // 0-based position of the dimension in the representation.
+  static constexpr std::int64_t Representation(
+      std::int64_t exponent,
+      std::int64_t position);
+
+ private:
+  static constexpr std::int64_t min_exponent = -24;
+  static constexpr std::int64_t max_exponent = 7;
+  static constexpr std::int64_t exponent_mask = 0x1F;
+  static constexpr std::int64_t exponent_bits = 5;
```

Comment:
Do we know why it is 5 and not 6? if so, document the historical reasons; perhaps we should migrate to 8 bits in the future.

Reply:
I have no idea.